core/jobs/batch_jobs/state_interaction_validation_jobs.py

```diff
@@ -0,0 +1,1382 @@
+                            f'State - {key} having RTE link tag do not '
+                            f'have url-with-value attribute'
+                        )
+
+            if len(rte_components_errors) > 0:
+                states_with_values.append(
+                    {
+                        'state_name': key,
+                        'rte_components_errors': rte_components_errors
+                    }
+                )
+        return states_with_values
+
+    @staticmethod
+    def _set_lower_and_upper_bounds(
+        range_var, lower_bound, upper_bound,
+        lb_inclusive, ub_inclusive
+    ) -> None:
+        """Sets the lower and upper bounds for the range_var, mainly
+        we need to set the range so to keep track if any other rule's
+        range lies in between or not to prevent redundancy
+
+        Args:
+            range_var: dict[str, Any]. To keep track of each rule's
+                ans group index, rule spec index, lower bound, upper bound,
+                lb inclusive, ub inclusive.
+
+            lower_bound: int. The lower bound.
+
+            upper_bound: int. The upper bound.
+
+            lb_inclusive: bool. If lower bound is inclusive.
+
+            ub_inclusive: bool. If upper bound is inclusive.
+        """
+        range_var['lower_bound'] = lower_bound
+        range_var['upper_bound'] = upper_bound
+        range_var['lb_inclusive'] = lb_inclusive
+        range_var['ub_inclusive'] = ub_inclusive
+
+    @staticmethod
+    def _is_enclosed_by(range_var, range_ele) -> bool:
+        """Checks whether the ranges of rules enclosed or not
+
+        Args:
+            range_var: dict[str, Any]. To keep track of each rule's
+                ans group index, rule spec index, lower bound, upper bound,
+                lb inclusive, ub inclusive.
+
+            range_ele: dict[str, Any]. To keep track of other rule's
+                ans group index, rule spec index, lower bound, upper bound,
+                lb inclusive, ub inclusive.
+
+        Returns:
+            is_enclosed: bool. Returns True if both rule's ranges are enclosed.
+        """
+        if (
+            range_ele['lower_bound'] is None or range_var['lower_bound'] is None
+            or range_ele['upper_bound'] is None or (
+                range_var['upper_bound'] is None)
+        ):
+            return False
+
+        lb_satisfied = (
+            range_ele['lower_bound'] < range_var['lower_bound'] or
+            (range_ele['lower_bound'] == range_var['lower_bound'] and (
+                not range_var['lb_inclusive'] or range_ele['lb_inclusive'])
+            )
+        )
+
+        ub_satisfied = (
+            range_ele['upper_bound'] > range_var['upper_bound'] or
+            (range_ele['upper_bound'] == range_var['upper_bound'] and (
+                not range_var['ub_inclusive'] or range_ele['ub_inclusive'])
+            )
+        )
+
+        is_enclosed = lb_satisfied and ub_satisfied
```

Comment:
This is quite hard to understand, especially because the variable names are almost the same (except for the last three characters). Can
```suggestion
        if (
            range_ele['lower_bound'] is None or 
            range_var['lower_bound'] is None or 
            range_ele['upper_bound'] is None or
            range_var['upper_bound'] is None
        ):
            return False

        lb_satisfied = (
            range_ele['lower_bound'] < range_var['lower_bound'] or
            (
                range_ele['lower_bound'] == range_var['lower_bound'] and
                (
                    not range_var['lb_inclusive'] or 
                    range_ele['lb_inclusive']
                )
            )
        )

        ub_satisfied = (
            range_ele['upper_bound'] > range_var['upper_bound'] or
            (
                range_ele['upper_bound'] == range_var['upper_bound'] and 
                (
                    not range_var['ub_inclusive'] or 
                    range_ele['ub_inclusive']
                )
            )
        )

        is_enclosed = lb_satisfied and ub_satisfied
```

Reply:
Oh yes, sorry about that.
Will `range_compare_to` and `range_compare_with` works? It's a bit descriptive. Please do let me know if I should change this.
Thanks

Reply:
compare to/with is not really descriptive either. I still don't understand the difference between the two.

Reply:
Oh okay,
So just to give some context to check if the two rules are not getting enclosed by we are doing this. To take an example suppose we have two rules 
1. If the number is less than or equal to 5. -> Some response
2. If the number is between 3 and 5. -> Some response

So if we take a look then the 2 rule will never be called as it will be made redundant by the 1 rule, so to track the ranges I'm using a range variable.
Now is_enclosed_by checks the range part if it is enclosed or not, simply the `range_compare_to` is the 2 rule and `range_compare_with` is the 1 rule. So just to keep track of all previous rules we have a list of range variables and we are calling `is_enclosed_by` for each previous rule.

I was not sure how to explain this part there, sorry about that.
Thanks